samples/apps/bookinfo/src/docker-compose.yaml

```diff
@@ -0,0 +1,178 @@
+version: '2'
+services:
+  etcd:
+    image: quay.io/coreos/etcd:latest
+    networks:
+      envoymesh:
+        aliases:
+          - etcd
+    ports:
+      - "4001:4001"
+      - "2380:2380"
+      - "2379:2379"
+    environment:
+      - SERVICE_IGNORE=1
+    command: ["/usr/local/bin/etcd", "-advertise-client-urls=http://0.0.0.0:2379", "-listen-client-urls=http://0.0.0.0:2379"]
+
+  istio-apiserver:
+    image: gcr.io/google_containers/kube-apiserver-amd64:v1.7.3
+    networks:
+      envoymesh:
+        ipv4_address: 172.28.0.13
+        aliases:
+          - apiserver
+    ports:
+      - "8080:8080"
+    privileged: true
+    environment:
+      - SERVICE_IGNORE=1
+    command: ["kube-apiserver", "--etcd-servers", "http://etcd:2379", "--service-cluster-ip-range", "10.99.0.0/16", "--insecure-port", "8080", "-v", "2", "--insecure-bind-address", "0.0.0.0"]
+
+  consul:
+    image: gliderlabs/consul-server
+    networks:
+      envoymesh:
+        aliases:
+          - consul
+    ports:
+      - "8500:8500"
+      - "172.28.0.1:53:8600/udp"
+      - "8400:8400"
+    environment:
+      - SERVICE_IGNORE=1
+    command: ["-bootstrap"]
+
+  registrator:
+    image: gliderlabs/registrator:latest
+    networks:
+      envoymesh:
+    volumes:
+      - /var/run/docker.sock:/tmp/docker.sock
+    command: ["-internal", "-retry-attempts=-1", "consul://consul:8500"]
+
+  discovery:
+    image: gihanson/discovery:latest
+    networks:
+      envoymesh:
+        aliases:
+          - istio-pilot
+    expose:
+      - "8080"
+    ports:
+      - "8081:8080"
+    volumes:
+      - ~/.kube/config:/istio/config
+
+  details-v1:
+    image: gihanson/details-v1:latest
+    networks:
+      envoymesh:
+    dns:
+      - 172.28.0.1
```

Comment:
which dns is this ?

can we generate this file from a template, without hardcoded private image repos?

Reply:
i think this is just the DNS for the consul  network.   Do we have any example in istio which generates yaml from template?

Reply:
there are too many :-) for instance istioctl kube-inject is one
the scary sed scripts in istio/istio/install/updateVersion.sh another
and I'm afraid there are 2 or 3 others around too

and there is also helm

is consul dns always running on 172.28.0.1 ? the network is probably customizable no ?

Reply:
Ok I've removed the hardcoded gihanson in the image namespace.   I don't think docker compose has a similar support like helm where you could template the yaml file, this is one of the top needed feature tho for compose.     @GregHanson can you chime in for the DNS?  my understanding is that is just a configurable variable however since it is private network for docker, this value can just work if user doesn't want to change it.

Reply:
@ldemailly 172.28.0.1 refers to the gateway defined at the end of the file for the `envoymesh` network.  So yes, this could be configured to any value - you would have to do the following:
for each of the containers, you need to set the corresponding dns field
```
    dns:
      - 172.28.0.1
```
change the port mapping for consul when it starts up
```
    ports:
       - "172.28.0.1:53:8600/udp"
```
and `apiserver` and `productpage-v1` have hardcoded IP addresses within the `subnet` defined for `envoymesh` network.  I opted to give them hardcoded IP addresses to simplify the demo steps since otherwise users would have to collect the various IP's every time for some of the steps
```
  istio-apiserver:
     networks:
       envoymesh:
         ipv4_address: 172.28.0.13
```

Reply:
Thank you @GregHanson!  Given consul set up took us a lot of time in the past, I recommend us leaving in the configuration as it is in the sample so user can have a good experience in running the sample, but put in a note that the envoymesh network is configurable for advanced consul and docker users.